src/c/luminehall.c

```diff
@@ -0,0 +1,305 @@
+#include "luminehall.h"
+#include "locs.h"
+
+void writeLumineHallText()
+{
+    //Function that writes out the Lumine text in the form of arrangements.
+    //It adds a VWF to what the base game did
+    byte *lumineText_curr_ptr = luminetext;
+    int *hallLineSize = &m2_hall_line_size;
+    int length = 0;
+    int currLenInTile = 0;
+    int currPos = 0;
+    int Tiles[4];
+    setTilesToBlank(Tiles);
+    //First things first, it calculates the total length of the text in pixels
+    while((*lumineText_curr_ptr) != END)
+    {
+        length += getCharWidth(*lumineText_curr_ptr);
+        lumineText_curr_ptr++;
+    }
+    //It then gets the length in arrangements. It also makes sure it's a number divisible by 8.
+    //This avoids having empty tiles showing
+    length = (length & 8 == 0 ? length : (length + (8 - (length & 7)))) >> 1;
+    (*hallLineSize) = length;
+    lumineText_curr_ptr = luminetext;
+    unsigned short *hallAddress = m2_get_hall_address();
+    //Prints out the characters
+    while((*lumineText_curr_ptr) != END)
+    {
+        readLumineCharacter(*lumineText_curr_ptr, Tiles, hallAddress, length, &currPos, &currLenInTile);
+        lumineText_curr_ptr++;
+    }
+    //Avoid having tiles that were not entirely printed
+    while((currPos*4) < length)
+    {
+        printVoidLumineTiles(hallAddress, length, currPos);
+        currPos++;
+    }
+}
+
+void readLumineCharacter(byte chr, int *Tiles, unsigned short *hallAddress, int length, int *currPos, int *currLen)
+{
+    //Reads a character. Handles special cases.
+    //The valid characters are printed to the Tiles 1bpp buffer that stores the VWF form of the text.
+    //This is then converted (when it makes sense to do so) to arrangements by printLumineTiles.
+    int tileWidth = 2;
+    int tileHeight = 2;
+    int chosenLen;
+    int renderedLen;
+    byte *glyphRows;
+    int AlternativeTiles[4];
+    switch(chr)
+    {
+        case END:
+            return;
+        case ENDL:
```

Comment:
Was there a specific reason for making an "ENDL" code? The original game just used [two spaces](https://github.com/Herringway/ebsrc/blob/8151d9b9521294f18d6da82846ceecc6f675f3bf/src/data/text/lumine_hall.asm) for that part, it looks like.

Reply:
Thought it was 8 pixels instead of 6. Fixing it.